indico/modules/events/registration/apple_pass.py

```diff
@@ -0,0 +1,133 @@
+        ticket.addSecondaryField('event-date', event_date, 'Date')
+        if self.event.address and self.event.has_location_info:
+            venue_name = (f'{self.event.room_name} ({self.event.venue_name})'
+                          if self.event.room_name and self.event.venue_name
+                          else (self.event.venue_name or self.event.room_name))
+            ticket.addSecondaryField('event-venue', venue_name, 'Venue')
+        ticket.addAuxiliaryField('registration-name', registration.full_name, 'Name')
+        ticket.addAuxiliaryField('registration-email', registration.email, 'Email')
+        ticket.addBackField('back-registration-name', registration.full_name, 'Name')
+        ticket.addBackField('back-event-venue', venue_name, 'Venue')
+        ticket.addBackField('back-ticket-number', f'#{registration.friendly_id}', 'Ticket number')
+        ticket.addBackField('back-event-date', event_date, 'Date')
+        ticket.addBackField('back-registration-email', registration.email, 'Email')
+        ticket.addBackField('back-event-url', f'<a href="{self.event.external_url}">Indico event page</a>', 'Link')
+        reg_url = url_for('event_registration.display_regform', registration.locator.uuid, _external=True)
+        ticket.addBackField('back-registration-details', f'<a href="{reg_url}">Registration details</a>', 'Link')
+
+        signals.event.registration.apple_pass_ticket_object.send(self.event, obj=ticket)
+        return ticket
+
+    def build_pass_object(self, registration):
+        from indico.modules.events.registration.util import get_persons, get_ticket_qr_code_data
+
+        # Extract Certificate details from certificate.pem itself
+        self.cert = x509.load_pem_x509_certificate(self.settings['apple_pass_certificate'].encode('UTF-8'))
+        cert_details = {d.split('=')[0]: d.split('=')[1] for d in self.cert.subject.rfc4514_string().split(',')}
+        passfile = IndicoPass(self.build_ticket_object(registration), passTypeIdentifier=cert_details['UID'],
+                              organizationName=cert_details['O'], teamIdentifier=cert_details['OU'])
+        passfile.logoText = cert_details['O']  # Add Organization name at the top
+        passfile.backgroundColor = '#007cac'
+        passfile.foregroundColor = passfile.labelColor = '#ffffff'
+        passfile.serialNumber = registration.ticket_uuid
+        qr_data = get_ticket_qr_code_data(get_persons([registration])[0])
+        passfile.barcode = Barcode(message=json.dumps(qr_data, separators=(',', ':')), format=BarcodeFormat.QR)
+
+        # Including the icon and logo is necessary for the passbook to be valid.
+        logo_path = config.ABSOLUTE_PASS_LOGO_PATH if config.ABSOLUTE_PASS_LOGO_PATH else (
+            os.path.join(self.static_path, 'images', 'logo_indico_small.png'))
+        passfile.addFile('icon.png', open(logo_path, 'rb'))  # noqa: SIM115
+        passfile.addFile('logo.png', open(logo_path, 'rb'))  # noqa: SIM115
```

Comment:
Why not using a context manager for these, rather than silencing the warning?

```suggestion
        with open(logo_path, 'rb') as f:
            passfile.addFile('icon.png', f)
            passfile.addFile('logo.png', f)
```

Reply:
oh, interesting, i wasn't aware that the logo here is a filesystem path and not a URL. yuck. I wonder if we should treat it as a URL in the config (and make an HTTP request to retrieve it), simply for the sake of consistency!

Reply:
@OmeGak I've tested you code: no error raised but it does not work, the logo is not added anymore.

Reply:
sounds like the library is as crappy as we all expected and expects the file to be still open after calling this? should be easy to verify by looking at the code of `addFile`...

Reply:
Done, commit: Fix icon - logo management